docs/pages/docs/guides/providers/openai.mdx

```diff
@@ -217,6 +210,47 @@ export default function Completion() {
 
 </Steps>
 
+## Guide: Handling Errors
+
+The OpenAI's API throws an `OpenAI.APIError` when an error occurs during a request. It is recommended to wrap your API calls in a `try/catch` block to handle these errors. For more information about `OpenAI.APIError`, see [OpenAI SDK Handling Errors](https://github.com/openai/openai-node?tab=readme-ov-file#handling-errors).
+
+```tsx filename="app/api/chat/route.ts"
+import OpenAI from 'openai'
+import { OpenAIStream, StreamingTextResponse } from 'ai'
+import { NextResponse } from 'next/server'
+
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+})
+
+export const runtime = 'edge'
+
+export async function POST(req: Request) {
+  // Wrap with a try/catch to handle API errors
+  try {
+    const { messages } = await req.json()
+
+    const response = await openai.chat.completions.create({
+      model: 'gpt-3.5-turbo',
+      stream: true,
+      messages,
+    })
+
+    const stream = OpenAIStream(response)
+
+    return new StreamingTextResponse(stream)
+  } catch (error) {
+    // Check if the error is an APIError
+    if (error instanceof OpenAI.APIError) {
+      const { name, status, headers, message } = error
+      return NextResponse.json({ name, status, headers, message }, { status })
```

Comment:
@MaxLeiter Not sure if it was a good idea to add NextResponse here.

Reply:
Yes because it's Next specific, but if it's alright for you then I'm good with it.

Reply:
Yeah let's use `new Response()` with JSON.stringify. Thanks for pointing this out!

Reply:
Um didn't resolve this yet but is it all good for now? Or I can fix it ASAP.